source/extensions/network/dns_resolver/apple/apple_dns_impl.cc

```diff
@@ -189,6 +203,7 @@ void AppleDnsResolverImpl::PendingResolution::onEventCallback(uint32_t events) {
     // Therefore, finish resolving with an error.
     pending_response_.status_ = ResolutionStatus::Failure;
     pending_response_.details_ = absl::StrCat("apple_dns_error_", error);
+    addTrace(static_cast<uint8_t>(AppleDnsTrace::Failed));
     finishResolve();
```

Comment:
Could we make finishResolve take the argument and add trace there?

Reply:
Done.